pretix_sepadebit/views.py

```diff
@@ -178,12 +179,16 @@ def get_queryset(self):
         ).order_by('-datetime')
 
     def get_unexported(self):
+        today = datetime.date.today()
+        latest_export_due_date = today + datetime.timedelta(days=int(self.request.event.settings.payment_sepadebit_prenotification_days))
```

Comment:
Hmm, this (and the one below) is a pretty breaking change, isn't it?

Previously, I was able to export payments more than ``prenotification_dates`` in the future. If my pre-notification days are set to 2 and I export every Monday, a payment created on a Sunday would now arrive at the bank 1 week later due to this change, correct?

Worse even, this basically forces me to do *daily* exports if I don't want to always *miss* the announced date, doesn't it? Most banks can handle just fine if you give them a file that contains debits a few days into the future.

I understand this isn't practical if your debit days is *months* in the future, but it might need an option at export time to handle it?

Reply:
Yeah, looking back I've made some bad assumptions here.
But I am not sure if I am understanding your point.

Was there a situation where there exist payments far out in the future? The due_date is computed as `day of order + prenotification_days`. Therefore there can exist only unexported payments in the past and up to `day of export + prenotification_days` (small caveat, my implementation needs an unlimited interval to include missed exports).
The now introduced upper limit should only cut off payments with far out due_dates. <details>
  <summary>Sidenote</summary>
This implementation mixes two different aspects of [SEPA direct debit timing (P. 31)](https://www.europeanpaymentscouncil.eu/sites/default/files/kb/file/2020-04/EPC016-06%202019%20SDD%20Core%20Rulebook%20version%201.1.pdf) . The pre-notification time (PT-04.02 - Notification of the customer) and the lead time for handing over the mandates to the bank (PT-04.03). It sounds like this was unproblematic up to this point and using my upper limit should keep it the same.
</details>

Aren't daily exports in this case not already mandatory?
Starting with a new event without any placed orders and a 2 days of prenotification time.
The Sunday order is therefore due on Tuesday. The Monday export contains this order, because it's due_date-range starts Monday and ends Wednesday. 
Every order placed after the export on Monday and the next Friday will miss their execution date, because the bank won't see them until the next Monday, when we export again. 
I think I always have to export payments on a shorter interval than my prenotification time to not miss due dates.

Reply:
You are correct, I was thinking wrong. Go ahead, then! :)


> (small caveat, my implementation needs an unlimited interval to include missed exports).

yup, that would be important.